src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -430,6 +430,10 @@ public async Task<AuthenticationResult> GetAuthenticationResultForAppAsync(
             if (string.IsNullOrEmpty(tenant))
             {
                 tenant = mergedOptions.TenantId;
+                if (!string.IsNullOrEmpty(tenant) && _metaTenantIdentifiers.Contains(tenant!) && !string.IsNullOrEmpty(mergedOptions.AppHomeTenantId))
```

Comment:
this class is tricky to test, what are your plans for testing?

Reply:
Maybe we could just extract the logic to compute the effective tenant (including the lines above and below) in an internal method, and write a unit test for this method?